### PR TITLE
FEAT: Implemented auth

### PR DESCRIPTION
**Pull request by shivanigangadharan**
Functions :- 

- Sign up
- Login
- Logout

### CODE REVIEW

File: src/context/authContext.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export const AuthProvider = ({ children }) => {
+
+    const [user, setUser] = useState(null);
+    const [encodedToken, setEncodedToken] = useState();
+
+    const SignupUser = async (email, password) => {
+        try {
+            const res = await axios.post(`/api/auth/signup`, {
+                firstName: "Shivani",
+                lastName: "Gangadharan",
```
Comment (Richa-rani02) [inline src/context/authContext.js]: Why r u assigning value while sending data it will come from input box  correct me if I misunderstood anything

**Review by Richa-rani02 — APPROVED** [review]
Rest everything looks fine liked that u have used alert and did validations too

File: src/context/authContext.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+const AuthContext = createContext();
+
+export const useAuth = () => useContext(AuthContext);
+
+export const AuthProvider = ({ children }) => {
+
+    const [user, setUser] = useState(null);
+    const [encodedToken, setEncodedToken] = useState();
+
+    const SignupUser = async (email, password) => {
```
Comment (Richa-rani02) [inline src/context/authContext.js]: While signup you have to send first name and last name inside async as well. You are sending only email and password

**Review by Richa-rani02 — CHANGES_REQUESTED** [review]

File: src/context/authContext.js (hunk 2 above)
Comment (shivanigangadharan) [reply src/context/authContext.js]: Done in commit [5d44d50](https://github.com/shivanigangadharan/streambucket/pull/1/commits/5d44d50992faa27cf26e6bbf0724535aba5b7a15).

File: src/context/authContext.js (hunk 1 above)
Comment (shivanigangadharan) [reply src/context/authContext.js]: You're right, thanks for pointing it out. Fixed it [5d44d50](https://github.com/shivanigangadharan/streambucket/pull/1/commits/5d44d50992faa27cf26e6bbf0724535aba5b7a15).

**Review by Richa-rani02 — APPROVED** [review]

**Pull request merged**